package/kedro_viz/api/rest/router.py

```diff
@@ -38,53 +38,48 @@ async def main():
     response_model_exclude_none=True,
 )
 async def get_single_node_metadata(node_id: str):
-    node = data_access_manager.nodes.get_node_by_id(node_id)
-    if not node:
-        return JSONResponse(status_code=404, content={"message": "Invalid node ID"})
-
-    if not node.has_metadata():
-        return JSONResponse(content={})
-
-    if isinstance(node, TaskNode):
-        return TaskNodeMetadata(node)
-
-    if isinstance(node, DataNode):
-        return DataNodeMetadata(node)
-
-    if isinstance(node, TranscodedDataNode):
-        return TranscodedDataNodeMetadata(node)
-
-    return ParametersNodeMetadata(node)
+    return get_node_metadata_response(node_id)
 
 
 @router.get(
     "/pipelines/{registered_pipeline_id}",
     response_model=GraphAPIResponse,
 )
 async def get_single_pipeline_data(registered_pipeline_id: str):
-    if not data_access_manager.registered_pipelines.has_pipeline(
-        registered_pipeline_id
-    ):
-        return JSONResponse(status_code=404, content={"message": "Invalid pipeline ID"})
-
-    modular_pipelines_tree = (
-        data_access_manager.create_modular_pipelines_tree_for_registered_pipeline(
-            registered_pipeline_id
+    return get_selected_pipeline_response(registered_pipeline_id)
+
+
+@router.post("/deploy")
+async def deploy_kedro_viz(input_values: S3DeployerCredentials):
+    try:
+        deployer = S3Deployer(input_values.region, input_values.bucket_name)
+        url = deployer.get_deployed_url()
+        response = {"message": "Website deployed on S3", "url": url}
+        return JSONResponse(status_code=200, content=response)
+    except PermissionError as exc:  # pragma: no cover
+        logger.exception("Permission error in deploying Kedro Viz : %s ", exc)
+        return JSONResponse(
+            status_code=401, content={"message": "Please provide valid credentials"}
+        )
+    except Exception as exc:  # pragma: no cover
+        logger.exception("Deploying Kedro Viz failed: %s ", exc)
+        return JSONResponse(
+            status_code=500, content={"message": "Failed to deploy Kedro Viz"}
         )
-    )
 
-    return GraphAPIResponse(
-        nodes=data_access_manager.get_nodes_for_registered_pipeline(  # type: ignore
-            registered_pipeline_id
-        ),
-        edges=data_access_manager.get_edges_for_registered_pipeline(  # type: ignore
-            registered_pipeline_id
-        ),
-        tags=data_access_manager.tags.as_list(),
-        layers=data_access_manager.get_sorted_layers_for_registered_pipeline(
-            registered_pipeline_id
-        ),
-        pipelines=data_access_manager.registered_pipelines.as_list(),
-        selected_pipeline=registered_pipeline_id,
-        modular_pipelines=modular_pipelines_tree,  # type: ignore
-    )
+
+@router.get(
+    "/package_compatibilities",
```

Comment:
In general, REST API routes should not contain special characters, the only exception being hyphen. It would be better if we call this as /package-compatibilities. 

However, I feel we should tackle this route differently if we check for compatibility in the backend. Something like a [POST] /feature-compatibility route which accepts a json body like 

```json
   {
       "feature": "DEPLOY_VIZ"
   }
```
and have some kind of dictionary in the backend for feature_package_compatibility like -

```python
feature_package_compatibility = {
  "DEPLOY_VIZ": {
     "fsspec": "2023.9.0" 
  }
}
```
This way we can check for feature compatibility and extend this if we need to check for multiple features in future. Since the frontend needs information on why some feature is not compatible, we can send specific message (containing the missing package dependency) or a generic message to the frontend like -

```json
{
"feature": "DEPLOY_VIZ",
"is_compatible": true,
"compatible_packages": {
        "fsspec": "2023.9.0" 
    }
}
```

Reply:
yup, already updated it to` package-compatibilities` in the FE PR and will also do the same in this PR. 

So for now it's only one package compatibility that we return but yes in future there might be more. In that case I was thinking we would return a List[PackageCompatibilityAPIResponse]. 

Also, at this point, I am not so sure if we want to make it a POST request having said that we can revise this in the future if the need arises, unless others feel strongly about this approach.

Reply:
Agree with both of you: this would be something we can scale going forward, though right now, without a use case to do so, I don't think we need to implement it at this stage.

Reply:
So i read this earlier today, and thought it seemed relevant here. Hence sharing - https://www.techtarget.com/whatis/definition/You-arent-gonna-need-it#:~:text=YAGNI%20principle%20(%22You%20Aren',desired%20increased%20frequency%20of%20releases.